crypto/rsa/rsa_pk1.c

```diff
@@ -138,7 +138,7 @@ int ossl_rsa_padding_add_PKCS1_type_2_ex(OSSL_LIB_CTX *libctx, unsigned char *to
     /* pad out with non-zero random data */
     j = tlen - 3 - flen;
 
-    if (RAND_bytes_ex(libctx, p, j, 0) <= 0)
+    if (j < 0 || RAND_bytes_ex(libctx, p, j, 0) <= 0)
```

Comment:
This `j<0` is impossible condition given the condition above at line 128 (RSA_PKCS1_PADDING_SIZE==11). As Coverity might flag it please do not add it.

Reply:
Hmm, and there should actually be check above for flen > 0 for correctness sake.

Reply:
done

Reply:
This assumes that tlen is properly sized?
if flen = 0 => j = tlen - 3.
if flen = 1 => j = tlen - 4
if flen = tlen - 11 => j = 8

Reply:
There is a condition that enforces tlen >= 11 before.

Reply:
More specifically, it raises an error if:
```c
flen > tlen - 11
```
with the added `flen >= 0` check, there is no need to check `j` here.